src/components/PrivateRoute.tsx

```diff
@@ -0,0 +1,22 @@
+import { Route, Redirect } from 'react-router-dom';
+import React from 'react';
+
+const PrivateRoute = ({ component: Component, ...rest }) => (
+  <Route
+    {...rest}
+    render={props => {
+      return rest.isAuthenticated ? (
```

Comment:
I'm curious as to why you chose to bundle `isAuthenticated` up into `rest`, instead of having `isAuthenticated: Boolean` in the component arguments?

Reply:
Mostly due to the sample I found doing it this way. But I think it would be better to explicitly have the isAuthenticated boolean in the component arguments. Will update.